migrations/versions/42cbae986650_policy_id_and_revoke.py

```diff
@@ -83,29 +96,39 @@ def downgrade():
     # get the list of existing policies from Arborist
     if not config["LOCAL_MIGRATION"]:
         existing_policies = list_policies(arborist_client, expand=True)
-
     # add the `resource_path` column back
     op.add_column("requests", sa.Column("resource_path", sa.String()))
 
     # get the list of existing policy_ids in the database
     connection = op.get_bind()
-    results = connection.execute("SELECT policy_id FROM requests").fetchall()
-    existing_policy_ids = set(r[0] for r in results)
-
-    for policy_id in existing_policy_ids:
-        if not config["LOCAL_MIGRATION"]:
-            resource_paths = get_resource_paths_for_policy(
-                existing_policies["policies"], policy_id
+    offset = 0
+    limit = 500
+    query = f"SELECT request_id, policy_id FROM requests ORDER by request_id LIMIT {limit} OFFSET {offset}"
+    results = connection.execute(query).fetchall()
+    while results:
+        for r in results:
+            request_id, policy_id = r[0], r[1]
+
+            if not config["LOCAL_MIGRATION"]:
+                resource_paths = get_resource_paths_for_policy(
+                    existing_policies["policies"], policy_id
+                )
+                assert (
+                    len(resource_paths) > 0
+                ), f"No resource_paths for policy {policy_id}"
+            else:
+                # hardcoded to avoid querying Arborist
+                resource_paths = ["/test/resource/path"]
+            # use the first item in the policy’s list of resources, because this
+            # schema only allows 1 resource_path
+            connection.execute(
+                f"UPDATE requests SET resource_path='{escape(resource_paths[0])}' WHERE request_id='{request_id}'"
             )
-            assert len(resource_paths) > 0, f"No resource_paths for policy {policy_id}"
-        else:
-            # hardcoded to avoid querying Arborist
-            resource_paths = ["/test/resource/path"]
-        # use the first item in the policy’s list of resources, because this
-        # schema only allows 1 resource_path
-        connection.execute(
-            f"UPDATE requests SET resource_path='{escape(resource_paths[0])}' WHERE policy_id='{escape(policy_id)}'"
-        )
+
+        # Grab another batch of rows
+        offset += limit
+        query = f"SELECT request_id, policy_id FROM requests ORDER by request_id LIMIT {limit} OFFSET {offset}"
```

Comment:
in both the upgrade and the downgrade migrations, we were making 1 DB call per policy ID; now we make one per row. Performance wise i think it may be better to go back to using `existing_policy_ids`, even if we have to do it once per batch

Reply:
this comment is still valid

Reply:
I'm in favor of performance. I've got a new commit where the queries have been reverted back to using policy_id and request_path instead of request_id.